### ISSUE TITLE
Unable to enter interactive command with a space if another command exists that starts with the same characters

### ISSUE THREAD

**Issue by lokedhs** [McCLIM/McCLIM#791]
In the example code, two commands are registered: `To Lisp` and `Toggle Foo`. It is impossible to enter the former command, as typing a space will be prevented because the latter command exists (which also starts with `To`).

Example code:

```
(eval-when (:compile-toplevel :load-toplevel :execute)
  (unless (find-package "CLIM")
    (ql:quickload "log4cl")
    (sb-ext:restrict-compiler-policy 'safety 3)
    (sb-ext:restrict-compiler-policy 'debug 3)
    (ql:quickload "mcclim")))

(defpackage :clim-test.template
  (:use :cl)
  (:export #:open-foo-frame))

(in-package :clim-test.template)

(clim:define-application-frame foo-frame ()
  ()
  (:panes (interaction-pane :interactor
                            :text-margins '(:left (:absolute 10)
                                            :right (:relative 10))))
  (:layouts (default (clim:vertically ()
                       interaction-pane)))
  (:geometry :width 600 :height 600))

(defun open-foo-frame ()
  (let ((frame (clim:make-application-frame 'foo-frame)))
    (clim:run-frame-top-level frame)))

(define-foo-frame-command (cmd-to-lisp :name "To Lisp")
    ()
  (let ((stream (clim:find-pane-named clim:*application-frame* 'interaction-pane)))
    (clim:with-room-for-graphics (stream :first-quadrant nil)
      (clim:surrounding-output-with-border (stream :ink clim:+blue+)
        (format stream "To Lisp")))))

(define-foo-frame-command (cmd-toggle-foo :name "Toggle Foo")
    ()
  (let ((stream (clim:find-pane-named clim:*application-frame* 'interaction-pane)))
    (clim:with-room-for-graphics (stream :first-quadrant nil)
      (clim:surrounding-output-with-border (stream :ink clim:+blue+)
        (format stream "Toggle")))))
```

**Comment by dkochmanski** [comment]
I have the same problem (confirmed).

**Comment by kc-** [comment]
Part of the issue is that #\Space gets passed as a `partial-completer` to the completion functions, see for example https://github.com/mcclim/mcclim/blob/master/Core/clim-core/commands.lisp#L1250.

The function `complete-input` in `input-editing.lisp` contains a comment about this (#\Space being a partial completer and a delimiter): https://github.com/mcclim/mcclim/blob/master/Core/clim-core/commands.lisp#L718 -- but that block is only entered when `nmatches` is 0, which does not apply in this example.

There's a debugging functionality in `input-editing.lisp`, you can watch part of what's happening by setting `*trace-complete-input*`:
```
(let ((climi::*trace-complete-input* T))
            (open-foo-frame))
```

We've discussed this briefly on IRC, and the preliminary consensus is that TAB should suffice as complete key. 
@dkochmanski Do you agree with this?

**Comment by lokedhs** [comment]
I agree with this proposal as well. It's not very clear how a scheme could be envisioned where space is a partial completer while at the same time everything works correctly.

**Comment by dkochmanski** [comment]
yes, I find tab more consistent with other toolkits I'm using.

**Comment by kc-** [comment]
The PR contains my proposal which I'm not very sure about. @lokedhs maybe you could try this out and opine on whether it changes the behaviour in the way you want?

**Comment by dkochmanski** [comment]
https://gist.github.com/dkochmanski/276b4a28c8ce176c0a2bd8af4dace96c here are my thoughts on the issue. I'm open to discussion about the desired behavior.

While removing space as partial completer is allowed we can't remove it as the command name delimiter. It is clearly stated in the spec, that default command name delimiter is space.

**Comment by dkochmanski** [comment]
markdown renderer does not display correctly table with test cases (it doesn't seem to notice the row-group separator), so to read them one should either look at raw file or copy said raw file to org-mode buffer in emacs

https://gist.githubusercontent.com/dkochmanski/276b4a28c8ce176c0a2bd8af4dace96c/raw/2f6ff893f870a3c0c75958f6fb941ed26070209f/foo.org
